src/content/blog/summer-of-codegen.mdx

```diff
@@ -0,0 +1,440 @@
+* [Claude Projects](): build interactive websites and other code through conversation. Websites can be viewed directly.
+* [Websim](): similar to Claude Projects, but built around "fake websites". So going to catgifs.com will generate a website with cat gifs on the fly.
+* [VSCode Copilot](): intelligent autocompletions of code within larger projects, for lots of programming languages.
+* [Cursor](): code editor with lots of AI: not just autocompletions but conversational code generation.
+
+<figure style={{ maxWidth: "900px", display: 'flex', justifyContent: 'space-between' }}>
+  <div style={{ maxWidth: "45%" }}>
+    <Image src={ClaudeImage} alt="" />
+    <figcaption>
+      Claude Projects: making a game.
+      {/* TODO: make gif? */}
+    </figcaption>
+  </div>
+  <div style={{ maxWidth: "53%" }}>
+    <Image src={WebsimImage} alt="" />
+    <figcaption>
+      Websim: catgifs.com.
+      {/* TODO: make gif? */}
+    </figcaption>
+  </div>
+</figure>
+
+The beauty of products like Vercel v0, Claude Projects, and Websim is that you immediately get a working program back even if you don't know anything about programming! For example, in Websim you don't even see the code, you just see the generated website.
+
+So far these LLM-generated programs have been limited to the frontend or copied and pasted into a another placements. Tools like VSCode Copilot and Cursor let you build larger software with a frontend and a backend, but now you need a place to deploy it, which is a significant barrier to non-programmers, and slows the iteration cycle.
+
+We se many users generate their code in an LLM and then copy it into Val Town, since our "vals" (tiny full-stack webapps) get instantly deployed, and can have a **frontend, backend, and database**.
+
+We figured we're in a unique position to tighten the feedback loop of codegen to full-stack deployment, to finally approach [end-user programming](/blog/end-programmer-programming):
+
+> The dream is that the full power of computers be **accessible to all**, that our
+digital worlds will be as **moldable as clay**.
+
+{/* <figure style={{ maxWidth: "600px" }}>
+  <Image src={AnnotatedValImage} alt="" />
+  <figcaption>
+    A "val" in Val Town: frontend, backend, database; immediately deployed.
+  </figcaption>
+</figure> */}
+
+I spent the last few weeks prototyping codegen ideas in Val Town, with great results. In this blog post I'll show you what I built and what I learned. And since I built it all within Val Town itself, it's all open source, and you can fork it!
+
+We're now full steam ahead on productionizing this within the product, so this blog post will also give you a sneak peak at what you can expect to see soon.
+
+<figure style={{ maxWidth: "600px" }}>
+  <video controls loop>
+    <source src="/video/make-real-with-backend.mp4" />
+  </video>
+  <figcaption>
+    "Make Real with Backend": one of the prototypes I made.
+  </figcaption>
+</figure>
+
+# Basic codegen
```

Comment:
Sadly I think we may want to delete this whole section. It read as a tutorial that's really in the weeds 

Dan G says it best: https://gist.github.com/stevekrouse/22d96b6142fd21a8f9b63500ebd2ee24#file-hn-advice-txt-L201-L300